File: scripts/audit_ja_sections.py
#!/usr/bin/env python3
import subprocess
import json
import re

programs = ['diet', 'digestive', 'head', 'neuro', 'pain', 'pediatric', 'skin', 'wellness', 'women']

def run_query(sql):
    cmd = f'''export PATH=$PATH:/usr/local/bin && npx wrangler d1 execute brd-clinic-db --remote --command "{sql}" 2>/dev/null'''
    result = subprocess.run(cmd, shell=True, capture_output=True, text=True, cwd='/Users/amu/dev/brd-clinic')
    return result.stdout

def extract_sections(output):
    try:
        # Find the JSON part
        match = re.search(r'\{[\s\S]*\}', output)
        if match:
            data = json.loads(match.group())
            if data.get('results') and len(data['results']) > 0:
                sections_str = data['results'][0].get('sections', '[]')
                return json.loads(sections_str)
    except:
        pass
    return []

print("=" * 80)
print("프로그램 섹션 비교 감사 (한국어 vs 일본어)")
print("=" * 80)

all_issues = []

for prog in programs:
    print(f"\n{'─' * 60}")
    print(f"프로그램: {prog.upper()}")
    print(f"{'─' * 60}")
    
    # Get Korean sections
    ko_output = run_query(f"SELECT sections FROM programs WHERE id = '{prog}'")
    ko_sections = extract_sections(ko_output)
    
    # Get Japanese sections
    ja_output = run_query(f"SELECT sections FROM program_translations WHERE program_id = '{prog}' AND locale = 'ja'")
    ja_sections = extract_sections(ja_output)
    
    print(f"\n한국어 섹션 ({len(ko_sections)}개):")
    for i, s in enumerate(ko_sections):
        stype = s.get('type', 'Unknown')
        title = (s.get('title') or '').replace('<br/>', ' ').replace('\n', ' ')[:30]
        img = s.get('image', '')
        img_name = img.split('/')[-1] if img else '(no image)'
        print(f"  {i+1}. [{stype}] {title}... 📷{img_name}")
    
    print(f"\n일본어 섹션 ({len(ja_sections)}개):")
    for i, s in enumerate(ja_sections):
        stype = s.get('type', 'Unknown')
        title = (s.get('title') or '').replace('<br/>', ' ').replace('\n', ' ')[:30]
        img = s.get('image', '')
        img_name = img.split('/')[-1] if img else '(no image)'
        print(f"  {i+1}. [{stype}] {title}... 📷{img_name}")
    
    # Compare
    issues = []
    
    if len(ko_sections) != len(ja_sections):
        issues.append(f"⚠️ 섹션 개수 불일치: 한국어 {len(ko_sections)}개 vs 일본어 {len(ja_sections)}개")
    
    max_len = max(len(ko_sections), len(ja_sections))
    for i in range(max_len):
        ko = ko_sections[i] if i < len(ko_sections) else None
        ja = ja_sections[i] if i < len(ja_sections) else None
        
        if ko and not ja:
            issues.append(f"❌ {i+1}번 [{ko.get('type')}]: 일본어에서 누락됨")
        elif ja and not ko:
            issues.append(f"⚠️ {i+1}번 [{ja.get('type')}]: 한국어에 없는 추가 섹션")
        elif ko and ja:
            if ko.get('type') != ja.get('type'):
                issues.append(f"❌ {i+1}번: 타입 불일치 - 한국어 [{ko.get('type')}] vs 일본어 [{ja.get('type')}]")
            if ko.get('image') != ja.get('image'):
                issues.append(f"⚠️ {i+1}번 [{ko.get('type')}]: 이미지 경로 불일치")
                issues.append(f"    한국어: {ko.get('image', '없음')}")
                issues.append(f"    일본어: {ja.get('image', '없음')}")
    
    print("\n🔍 비교 결과:")
    if issues:
        for issue in issues:
            print(f"  {issue}")
        all_issues.append((prog, issues))
    else:
        print("  ✅ 모든 섹션 일치")

print("\n" + "=" * 80)
print("요약")
print("=" * 80)

if all_issues:
    print(f"\n문제가 있는 프로그램: {len(all_issues)}개")
    for prog, issues in all_issues:
        print(f"\n  📛 {prog}:")
        for issue in issues:
            print(f"    {issue}")
else:
    print("\n✅ 모든 프로그램의 일본어 섹션이 한국어와 일치합니다.")

File: scripts/audit_ja_sections.js
// Script to audit Japanese program sections against Korean source
const programIds = ['diet', 'digestive', 'head', 'neuro', 'pain', 'pediatric', 'skin', 'wellness', 'women'];

async function main() {
    const dbPath = '.wrangler/state/v3/d1/miniflare-D1DatabaseObject/27800fe4107c459a7c19f03056d1618183625b44e7ba01d6a175dd53d683ab5a.sqlite';
    const Database = require('better-sqlite3');
    const db = new Database(dbPath);

    console.log('='.repeat(80));
    console.log('日本語プログラムセクション 監査レポート');
    console.log('='.repeat(80));
    console.log('');

    for (const programId of programIds) {
        console.log(`\n${'─'.repeat(60)}`);
        console.log(`프로그램: ${programId.toUpperCase()}`);
        console.log(`${'─'.repeat(60)}`);

        // Get Korean sections (source of truth)
        const koRow = db.prepare("SELECT sections FROM programs WHERE id = ?").get(programId);
        const koSections = koRow?.sections ? JSON.parse(koRow.sections) : [];

        // Get Japanese sections
        const jaRow = db.prepare("SELECT sections FROM program_translations WHERE program_id = ? AND locale = 'ja'").get(programId);
        const jaSections = jaRow?.sections ? JSON.parse(jaRow.sections) : [];

        console.log(`\n한국어 섹션 (${koSections.length}개):`);
        koSections.forEach((s, i) => {
            const img = s.image ? `📷 ${s.image.split('/').pop()}` : '(이미지없음)';
            console.log(`  ${i + 1}. [${s.type}] ${s.title?.replace(/<br\/?>/g, ' ').substring(0, 30)}... ${img}`);
        });

        console.log(`\n일본어 섹션 (${jaSections.length}개):`);
        jaSections.forEach((s, i) => {
            const img = s.image ? `📷 ${s.image.split('/').pop()}` : '(이미지없음)';
            console.log(`  ${i + 1}. [${s.type}] ${s.title?.replace(/<br\/?>/g, ' ').substring(0, 30)}... ${img}`);
        });

        // Compare
        console.log('\n🔍 비교 결과:');

        // Check count
        if (koSections.length !== jaSections.length) {
            console.log(`  ⚠️  섹션 개수 불일치: 한국어 ${koSections.length}개 vs 일본어 ${jaSections.length}개`);
        } else {
            console.log(`  ✅ 섹션 개수 일치: ${koSections.length}개`);
        }

        // Check order and types
        const maxLen = Math.max(koSections.length, jaSections.length);
        const orderIssues = [];
        const imageIssues = [];

        for (let i = 0; i < maxLen; i++) {
            const ko = koSections[i];
            const ja = jaSections[i];

            if (!ko) {
                orderIssues.push(`  ⚠️  ${i + 1}번: 한국어에 없는 추가 섹션 [${ja?.type}]`);
            } else if (!ja) {
                orderIssues.push(`  ❌ ${i + 1}번: 일본어에 누락된 섹션 [${ko?.type}]`);
            } else if (ko.type !== ja.type) {
                orderIssues.push(`  ❌ ${i + 1}번: 타입 불일치 - 한국어 [${ko.type}] vs 일본어 [${ja.type}]`);
            }

            // Check images
            if (ko && ja && ko.image !== ja.image) {
                imageIssues.push(`  ❌ ${i + 1}번 [${ko.type}]: 이미지 불일치\n      한국어: ${ko.image || '없음'}\n      일본어: ${ja.image || '없음'}`);
            }
        }

        if (orderIssues.length > 0) {
            console.log('\n  📋 순서/타입 문제:');
            orderIssues.forEach(issue => console.log(issue));
        } else {
            console.log('  ✅ 섹션 순서 일치');
        }

        if (imageIssues.length > 0) {
            console.log('\n  🖼️  이미지 문제:');
            imageIssues.forEach(issue => console.log(issue));
        } else {
            console.log('  ✅ 이미지 경로 일치');
        }
    }

    console.log('\n' + '='.repeat(80));
    console.log('감사 완료');
    console.log('='.repeat(80));

    db.close();
}

main().catch(console.error);
